core/src/main/java/fr/sncf/osrd/envelope_sim/IntegrationStep.java

```diff
@@ -34,10 +34,10 @@ public static IntegrationStep fromNaiveStep(
         // if the end of the step dips below 0, cut the step in half
         if (endSpeed < 0.0) {
             assert directionSign * acceleration < 0.0;
-            timeDelta = -startSpeed / (directionSign * acceleration);
+            endSpeed = 0.0;
+            timeDelta = (endSpeed - startSpeed) / (directionSign * acceleration);
```

Comment:
Why using `endSpeed` since it's value is `0.`?
```suggestion
            timeDelta = -startSpeed / (directionSign * acceleration);
```

Reply:
We thought with Alex that it is clearer to use the general formula timeDelta = (endSpeed - startSpeed) / (directionSign * acceleration) and to specify that in this case endSpeed is equal to zero. Don't you agree?

Reply:
solved by https://github.com/DGEXSolutions/osrd/pull/636/commits/139bdf8dee285ea7372b6109bbc4d49bcb39599d